x/deployment/ante/ante.go

```diff
@@ -0,0 +1,47 @@
+package ante
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+
+	"github.com/ovrclk/akash/x/deployment/keeper"
+	"github.com/ovrclk/akash/x/deployment/types"
+)
+
+type DeploymentDecorator struct {
+	keeper keeper.Keeper
+}
+
+// NewHandler returns a handler for "deployment" type messages
+func NewDeploymentDecorator(keeper keeper.Keeper) DeploymentDecorator {
+	return DeploymentDecorator{
+		keeper: keeper,
+	}
+}
+
+func (dd DeploymentDecorator) AnteHandle(ctx sdk.Context, tx sdk.Tx, simulate bool, next sdk.AnteHandler) (sdk.Context, error) {
+	for _, msg := range tx.GetMsgs() {
+		switch msg := msg.(type) {
+		case types.MsgCreateDeployment:
```

Comment:
break these cases out into smaller functions like is done in the handler package.